src/stk/math/types.h

```diff
@@ -126,7 +127,39 @@ struct STK_ALIGN(16) double4
 };
 
 
-#endif
+#endif // STK_USE_CUDA
+
+
+template<typename T, unsigned int rows_, unsigned int cols_>
+class matrix
```

Comment:
We need a way to represent image direction. This is mostly a temporary solution, I think if we need matrix operations we should look into some serious LA backend.

Reply:
Hmm... Given that we probably only need a matrix for the direction, can't we simply create a Matrix3x3f type thing? This is both expensive and a bit overkill tbh.

Reply:
I started doing exactly that, but then I changed my mind, since usually when I start needing just one matrix, after a short while I realise I need a whole LA library.

Anyway you're right, a fully featured class with private state is more expensive than necessary here, considering I doubt we will expand this class further. I rewrote it as a POD struct (the barely minimum to be assignable) with just an inline accessor and an auxiliary method to simplify initialisation (to not have to write the full matrix or assign the components one by one to just set the diagonal).

However we will need some more LA machinery if we plan to actually use the direction matrix (for starters, to compute its inverse), so probably going custom here is not a wise idea.